main.go

```diff
@@ -14,7 +14,339 @@
+	}
+	if err := envconfig.Process("nsm", config); err != nil {
+		logrus.Fatalf("error processing config from env: %+v", err)
+	}
+
+	log.FromContext(ctx).Infof("Config: %#v", config)
+	log.FromContext(ctx).WithField("duration", time.Since(now)).Infof("completed phase 1: get config from environment")
+
+	log.FromContext(ctx).Infof("executing phase 2: retrieving svid, check spire agent logs if this is the last line you see (time since start: %s)", time.Since(starttime))
+	now = time.Now()
+	source, err := workloadapi.NewX509Source(ctx)
+	if err != nil {
+		logrus.Fatalf("error getting x509 source: %+v", err)
+	}
+	svid, err := source.GetX509SVID()
+	if err != nil {
+		logrus.Fatalf("error getting x509 svid: %+v", err)
+	}
+	logrus.Infof("SVID: %q", svid.ID)
+	log.FromContext(ctx).WithField("duration", time.Since(now)).Info("completed phase 2: retrieving svid")
+
+	log.FromContext(ctx).Infof("executing phase 3: create ovsxconnect network service endpoint (time since start: %s)", time.Since(starttime))
+
+	xConnectEndpoint, err := createInterposeEndpoint(ctx, config, source)
+	if err != nil {
+		logrus.Fatalf("error configuring forwarder endpoint: %+v", err)
+	}
+
+	log.FromContext(ctx).WithField("duration", time.Since(now)).Info("completed phase 3: create ovsxconnect network service endpoint")
+
+	log.FromContext(ctx).Infof("executing phase 4: create grpc server and register ovsxconnect (time since start: %s)", time.Since(starttime))
+	tmpDir, err := ioutil.TempDir("", "cmd-forwarder-ovs")
+	if err != nil {
+		log.FromContext(ctx).Fatalf("error creating tmpDir: %+v", err)
+	}
+	defer func() { _ = os.RemoveAll(tmpDir) }()
+	listenOn := &url.URL{Scheme: "unix", Path: path.Join(tmpDir, "listen_on.io.sock")}
+
+	server := registerGRPCServer(source, xConnectEndpoint)
+	srvErrCh := grpcutils.ListenAndServe(ctx, listenOn, server)
+	exitOnErrCh(ctx, cancel, srvErrCh)
+	log.FromContext(ctx).WithField("duration", time.Since(now)).Info("completed phase 4: create grpc server and register ovsxconnect")
+
+	// ********************************************************************************
+	log.FromContext(ctx).Infof("executing phase 5: register %s with the registry (time since start: %s)", config.NSName, time.Since(starttime))
+	// ********************************************************************************
+	err = registerEndpoint(ctx, config, source, listenOn)
+	if err != nil {
+		log.FromContext(ctx).Fatalf("failed to connect to registry: %+v", err)
+	}
+	log.FromContext(ctx).WithField("duration", time.Since(now)).Infof("completed phase 5: register %s with the registry", config.NSName)
+
+	log.FromContext(ctx).Infof("Startup completed in %v", time.Since(starttime))
+
+	<-ctx.Done()
+	<-srvErrCh
+}
+
+func getTermSigalContext() (ctx context.Context, stop context.CancelFunc) {
+	return signal.NotifyContext(
+		context.Background(),
+		os.Interrupt,
+		// More Linux signals here
+		syscall.SIGHUP,
+		syscall.SIGTERM,
+		syscall.SIGQUIT,
+	)
+}
```

Comment:
I'd counsel moving this to being inline, as it makes it clearer what's happening.  Wrapping a single function call in a function call reduces the clarity of what is going on.

Please note, this point is a matter of taste, so feel free to disagree with me here.  I offer this as advice, not as 'You need to fix this to get merged'.

Reply:
@edwarnicke agree with your point, have changed back using `signal.NotifyContext `as inline function call.